src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -4579,6 +4579,21 @@ internal enum AddressKind
                         stackLocalsOpt));
                     goto case BoundKind.ConditionalReceiver;
 
+                case BoundKind.ComplexReceiver:
```

Comment:
Can we reach this? I thought this node would only be introduced in a later phase (but not during initial binding).

Reply:
> Can we reach this? I thought this node would only be introduced in a later phase (but not during initial binding).

The same goes for ```BoundKind.ComplexConditionalReceiver```. Just follow the pattern.

Reply:
Why not throw unreachable exception since this is unreachable? I'd do that for `ComplexConditionalReceiver` too

Reply:
> Why not throw unreachable exception since this is unreachable? I'd do that for ```ComplexConditionalReceiver``` too

I am comfortable going as is